jwst/extract_1d/extract.py

```diff
@@ -1257,8 +1257,9 @@ def do_extract1d(input_model, refname, smoothing_length, bkg_order):
             # For this case, we'll call get_spectral_order to get the order.
             spectral_order_list = ["not set yet"]
 
-        if isinstance(input_model, datamodels.ImageModel) or \
-           isinstance(input_model, datamodels.DrizProductModel):
+        if isinstance(input_model, (datamodels.ImageModel,
+                                    datamodels.DrizProductModel,
+                                    datamodels.IFUImageModel)):
```

Comment:
I don't think extract_1d should ever be fed an IFUImageModel as input. IFUImageModel is a simple 2D image of the individual slices. Extract_1d can only work on an IFUCubeModel.